File: Solution/Aoc/Aoc.Domain/Compute/Instructions/InstructionTypes/Jump.cs
namespace Aoc.Domain.Compute.Instructions.InstructionTypes
{
    internal abstract class Jump : Instruction
    {
        public override int Length => 3;
        
        internal Jump(int opcode) : base(opcode)
        {
        }

        public abstract bool ShouldJump(int parameter);
    }
}

File: python/aoc/domain/compute/instructions/jump.py
from domain.compute.instructions.instruction import Instruction


class Jump(Instruction):
    @property
    def _jumpIfTrue(self):
        return self.__jumpIfTrue
    @property
    def Length(self):
        return 3
    def __init__(self, opcode, jumpIfTrue):
        super(Jump, self).__init__(opcode)
        self.__jumpIfTrue = jumpIfTrue
    def ShouldJump(self, parameter):
        if self.__jumpIfTrue():
            return (parameter > 0)
        return (parameter == 0)
